modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala

```diff
@@ -57,27 +59,43 @@ object Sequence {
     val next: Option[Zipper] =
       focus.next match {
         // Step completed
-        case None      =>
-          pending match {
+        case None      => {
+          remaining match {
```

Comment:
Look into `remaining` instead of `pending`, effectively skipping the steps in `toSkip`.